driver-core/src/main/java/com/datastax/driver/core/JdkSSLOptions.java

```diff
@@ -50,17 +53,36 @@ protected JdkSSLOptions(SSLContext context, String[] cipherSuites) {
         this.cipherSuites = cipherSuites;
     }
 
+    /**
+     * Creates an SSLHandler for a connection.
+     * You might want to override this (and newSSLEngine) if you need to fine-tune the engine's configuration
+     * (for example enabling hostname verification).
+     * 
+     * @param channel the Netty channel for that connection.
+     * @param address the remote address of the connection.
+     */
     @Override
-    public SslHandler newSSLHandler(SocketChannel channel) {
+    public SslHandler newSSLHandler(SocketChannel channel, InetSocketAddress address) {
         SSLEngine engine = newSSLEngine(channel);
         return new SslHandler(engine);
     }
 
     /**
+     * Creates an SSLHandler for a connection.
+     * Deprecated: code uses newSSLHandler(SocketChannel channel, InetSocketAddress address) instead.
+     * @param channel the Netty channel for that connection.
+     */
+    @Deprecated
+    @Override
+    public SslHandler newSSLHandler(SocketChannel channel) {
+        throw new UnsupportedOperationException("This is a placeholder, please use newSSLHandler(SocketChannel channel, InetSocketAddress address)");
+    }
+
+    /**
      * Creates an SSL engine each time a connection is established.
      * <p/>
      * <p/>
-     * You might want to override this if you need to fine-tune the engine's configuration
+     * You might want to override this (and newSSLHandler) if you need to fine-tune the engine's configuration
```

Comment:
nit: `(and its {@link #newSSLEngine(SocketChannel, InetSocketAddress) overloaded method})`